controllers/volumereplicationgroup_controller.go

```diff
@@ -498,17 +505,63 @@ func (s ObjectStorePVDownloader) DownloadPVs(ctx context.Context, r client.Reade
 	return objectStore.downloadPVs(s3Bucket)
 }
 
-func (v *VRGInstance) restorePV(pv *corev1.PersistentVolume) error {
-	v.cleanupPVForRestore(pv)
+func (v *VRGInstance) restorePVMetadata(pvList []corev1.PersistentVolume) error {
+	numRestored := 0
+
+	for idx := range pvList {
+		pv := &pvList[idx]
+		v.cleanupPVForRestore(pv)
+		v.addPVRestoreAnnotation(pv)
+
+		if err := v.reconciler.Create(v.ctx, pv); err != nil {
+			if errors.IsAlreadyExists(err) {
+				err := v.validatePVExistence(pv)
+				if err != nil {
+					v.log.Info("PV exists. Ignoring and moving to next PV", "error", err.Error())
+					// ignoring any errors
+					continue
```

Comment:
Do we need to increment numRestored in this case as well? (We probably can call it numPVsPresent to suit it better with the purpose).

The reason being, suppose restorin from a S3Profile fails (for some reason) and in the failed S3Profile we were able to restore a subset of the PVs. Now, due to failure we move on to the next S3Profile. There when we try to restore pv we first check that whether the PV already exists or not. If so, then we ignore any possible error (which can happen when the PV does not contain the restore annotation IIUC). But, since we dont increment numRestored variable, even though we end up in restoring all the PVs with this S3Profile, we consider to have failed (as numRestored != len(pvList).

Let me know if I am missing something.

Reply:
`validatePVExistence` will return success if we restored the PV, so when we move to the next S3 store, we will increment the `numRestored`. The only case we do not increase `numRestored` is if we find a PV name conflict that ramen did not restore, which is a hard error, as we do not want to overwrite a system created PV.

So if PV does not have the annotation it is an error, and we should not proceed.

Reply:
ACK